### ISSUE TITLE
Syntax highlighting is broken

### ISSUE THREAD

**Issue by vindarel** [LispCookbook/cl-cookbook#279]
Hello guys,

The syntax highlighting is broken. It worked today on my out-of-date gh-pages branch, and it doesn't anymore that I rebased on master.

**Comment by pnathan** [comment]
Not good.

**Comment by vindarel** [comment]
Well any idea is welcome. I just tried locally witht `jekyll serve` and I have syntax highlighting. Same output in the browser console.

**Comment by sheepduke** [comment]
I tested the latest master branch (4a8e828) on my machine with the following command:

```
bundle update
bundle exec jekyll serve
```

And it does **not** work.

**Comment by sheepduke** [comment]
By the way, just some random thoughts - I am using Hugo for my personal blog. It works well with both Markdown and Org. Yes, I can write Org file which will be rendered properly. And it is much faster.

I am not sure if you are interested with it...

**Comment by pcr910303** [comment]
@sheepduke While I’m also a big fan of org mode (which allows consistent technical documentation easier), the only platform org works is in Emacs. People not using Emacs shouldn’t be a second-class user/contributor. 
Markdown has a much universal support.

**Comment by sheepduke** [comment]
On 2019-12-02 06:36, 조성빈 <notifications@github.com> wrote:
> @sheepduke While I’m also a big fan of org mode (which allows consistent technical documentation easier), the only platform org works is in Emacs. People not using Emacs shouldn’t be a second-class user/contributor. 
> Markdown has a much universal support.
>
> -- 
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub:
> https://github.com/LispCookbook/cl-cookbook/issues/279#issuecomment-560423027

Well, you are right.

It is just the number of people who writes Common Lisp without Emacs is
rather minority.

Referring to the issue itself, it might because of missing highlight.js
file?

**Comment by vindarel** [comment]
> Referring to the issue itself, it might because of missing highlight.js
file?

Is it missing?? It shouldn't be, it was included before.

thanks for the report.

It's true that Hugo is much faster. I'd be in favor of this change (specially if we improve the look and feel in the process?).

**Comment by sheepduke** [comment]
On 2019-12-02 08:21, vindarel <notifications@github.com> wrote:
>> Referring to the issue itself, it might because of missing highlight.js
> file?
>
> Is it missing?? It shouldn't be, it was included before.
>
OK, I found the file.

It is placed here: `./assets/highlight-lisp.js`

I found that GitHub Pages is using jekyll-3.8.5 and our Gemfile declares
`gem 'jekyll', '>= 3.6.3'`.

https://pages.github.com/versions/

But if I change it to `gem 'jekyll', '>= 3.8.5'` and run `bundle exec
jekyll serve`, it still does not work.

But my global jekyll, which is also 3.8.5 version, works.

So I did some experiments:

             bundle        global jekyll
master        No              Yes
83238bc       No              Yes

It seems that it is the problem with bundle, not our configuration.

I have no idea how to fix it then. :-(

> It's true that Hugo is much faster. I'd be in favor of this change (specially if we improve the look and feel in the process?).
>
From my personal experience, I would say Hugo is much more convenient
compared to Jekyll.

I wrote my experience when setting up my blog.

http://sheepduke.com/posts/2019-05-11_use-hugo-to-setup-personal-blog/

Here is a list of themes with `documentation` tag:

https://themes.gohugo.io/tags/documentation/

**Comment by vindarel** [comment]
I could reproduce and break my local syntax highlighting by updating the gems (I had to delete Gem.lock and google a bit).

Now it seems that the generated HTML changed a bit. Code blocks are:

```
<pre class="lisp"
  <code>
```

it's `<pre>` that gets the class, not `<code>`.

HighlightLisp wants `<code>`s:

```
	this.highlight_auto = function(options)
	{
		options || (options = {});
		var classname = options.className ? options.className : 'lisp';
		var codes = document.getElementsByTagName('code');
		for(var i = 0, n = codes.length; i < n; i++)
		{
			var code = codes[i];
			if(code.className.match(classname))
			{
				this.highlight_element(code);
			}
		}
	},
```

A couple attempts of mine only messed things worst so far.

It's said that Jekyll supports CL highlighting with "fenced blocks" (aka triple backquotes), but I can't make them work either.

**Comment by vindarel** [comment]
I pinned Jekyll to 3.6.3, it works locally with bundle exe, but it doesn't to when pushed to gh-pages…

It seems that Jekyll 3.6.8 does the syntax highlighting of our code snippets, thus transforming them into inner html nodes, and as a consequence HighlightLisp cannot find strings to replace.

I think we have to disable the syntax highlighting, but I didn't find how.
https://jekyllrb.com/docs/configuration/default/

**Comment by arademaker** [comment]
Can’t we use another lisp static site generator?

**Comment by vindarel** [comment]
I guess we can.

(https://github.com/coleslaw-org/coleslaw)

**Comment by vindarel** [comment]
https://pages.github.com/versions/

**Comment by vindarel** [comment]
fixed by 61180f1cb0d9712caa8314f2802f6c681d04f36e

The highlighting is less good than HighlightLisp though.

**Comment by vindarel** [comment]
We're back with the good syntax highlighting.
